src/Verifier/SAW/FiniteValue.hs

```diff
@@ -59,6 +60,7 @@ data FirstOrderValue
   | FOVInt Integer
   | FOVWord Natural Integer -- ^ a more efficient special case for 'FOVVec FOTBit _'.
   | FOVVec FirstOrderType [FirstOrderValue]
+  | FOVArray FirstOrderType FirstOrderType
```

Comment:
The primary purpose of datatype `FirstOrderValue` is for representing concrete values from counterexamples. This declaration of `FOVArray` doesn't work for that. We could change this to something like
```
  FOVArray FirstOrderType FirstOrderValue [(FirstOrderValue, FirstOrderValue)]
```
to represent a constant array with a sequence of updates (the first argument is the domain type of the array, in case the list is empty).

Alternatively we could just leave the array type out of `FirstOrderType` and `FirstOrderValue` completely. It might be a while before we have any proper support for obtaining concrete array values from solver counterexamples (see GaloisInc/what4#39). Until we do, it might make sense to just leave out all these changes to `FiniteValue.hs` from this PR.

Reply:
I think the array type must be part of `FirstOrderType` in order to support constants/variables of array type and uninterpreted functions over arrays. `FirstOrderValue` is used to display counterexamples, and the counterexamples may have variables of array type, so I need to handle that case. This is essentially a dummy value that doesn't contain any information about the array, just that there is some array, and prevents `saw` from crushing when displaying counterexamples.

Reply:
This seems like a reasonable compromise: We can keep the `FOVArray` constructor with only type information, so that obtaining and displaying counterexamples will work, even though the actual values are not represented.

But as this is incurring a bit of tech debt, we should at least add a comment explaining that `FOVArray` is intentionally designed to include type information only for now (until such time that we have a way to obtain array values from SMT solvers).